File: src/main/java/xyz/emirdev/echogen/utils/LuckPermsUtils.java
package xyz.emirdev.echogen.utils;

import java.util.Collection;
import java.util.Objects;
import java.util.Set;

import org.bukkit.entity.Player;

import net.luckperms.api.model.group.Group;
import net.luckperms.api.model.user.User;
import net.luckperms.api.query.QueryOptions;
import xyz.emirdev.echogen.Echogen;

public class LuckPermsUtils {
    public static User getUser(Player player) {
        return Echogen.get().getLuckPerms().getUserManager().getUser(player.getUniqueId());
    }

    public static String getPrefix(Player player) {
        return Objects.requireNonNullElse(getUser(player).getCachedData().getMetaData().getPrefix(), "");
    }

    public static String getSuffix(Player player) {
        return Objects.requireNonNullElse(getUser(player).getCachedData().getMetaData().getSuffix(), "");
    }

    public static Collection<Group> getPlayerGroups(Player player) {
        return getUser(player).getInheritedGroups(QueryOptions.defaultContextualOptions());
    }

    public static Set<Group> getAllGroups() {
        return Echogen.get().getLuckPerms().getGroupManager().getLoadedGroups();
    }
}

File: src/main/kotlin/cat/emir/echogen/utils/LuckPermsUtils.kt
package cat.emir.echogen.utils

import org.bukkit.entity.Player

import net.luckperms.api.model.group.Group
import net.luckperms.api.model.user.User
import net.luckperms.api.query.Flag
import net.luckperms.api.query.QueryOptions
import cat.emir.echogen.Echogen

class LuckPermsUtils(val plugin: Echogen) {
    fun getUser(player: Player): User? {
        return plugin.luckPerms.userManager.getUser(player.uniqueId)
    }

    fun getPrefix(player: Player): String {
        return getUser(player)?.cachedData?.metaData?.prefix ?: ""
    }

    fun getSuffix(player: Player): String {
        return getUser(player)?.cachedData?.metaData?.suffix ?: ""
    }
    fun getPlayerGroups(player: Player): Set<Group> {
        val queryOptions = QueryOptions.defaultContextualOptions().toBuilder()
                .flag(Flag.RESOLVE_INHERITANCE, false)
                .build()
        return getUser(player)?.getInheritedGroups(queryOptions)?.toSet() ?: setOf()
    }

    fun getAllGroups(): Set<Group> {
        return plugin.luckPerms.groupManager.loadedGroups.toSet()
    }
}
